src/corona_hakab_model/analyzers/simulation_analysis.py

```diff
@@ -62,28 +65,92 @@ def plot_aggregations_from_csv(csv_file_path, aggregation_funcs: Dict[str, Calla
+        rects_list.append(ax.bar(x - width * (- n_aggregations + 2 * i) / 2, list(agg_values), width, label=agg_name))
 
     ax.set_title(f"aggregation of {csv_file_path.split('/')[-1]}")
     ax.set_xticks(x)
     ax.set_xticklabels(runs_labels)
     ax.legend()
 
-    def label_rects_with_hight(rects):
+    def label_rects_with_height(rects):
         """Attach a text label above each bar in *rects*, displaying its height."""
         for rect in rects:
-            height = rect.get_height()
+            height = round(rect.get_height(), 2)
             ax.annotate('{}'.format(height),
                         xy=(rect.get_x() + rect.get_width() / 2, height),
                         xytext=(0, 3),  # 3 points vertical offset
                         textcoords="offset points",
                         ha='center', va='bottom')
 
-    for rect in rects:
-        label_rects_with_hight(rect)
+    for rects in rects_list:
+        label_rects_with_height(rects)
 
     fig.tight_layout()
     plt.show()
 
+
+def create_time_avg_std(comparison_dir_path) -> Dict[str, pd.DataFrame]:
+    """
+    see example_script.py for usage
+    :param comparison_dir_path: path to the directory created by create_comparison_files to work on
+    :return: a dictionary with to keys: "mean", "std" with the compatible DataFrames as values
+    """
+    return create_time_step_aggregation(comparison_dir_path, agg_funcs_dict={"mean": np.mean, "std": np.std})
+
+
+def create_time_step_aggregation(parameters_dir_path, agg_funcs_dict: Dict[str, Callable]) -> Dict[str, pd.DataFrame]:
+    """
+
+    :param parameters_dir_path: path to the directory created by create_comparison_files to work on
+    :param agg_funcs_dict: dict of all the aggregation functions that want to be applied across multiple simulation runs
+    :return: a dictionary with keys the same as agg_funcs_dict but values the df corresponding to the aggregation
+    """
+    files = glob(os.path.join(OUTPUT_FOLDER, parameters_dir_path, "*.csv"))
+    dfs_dict = dict(map(lambda f: (os.path.basename(f),
+                                   pd.read_csv(f).drop(["Unnamed: 0"], axis=1, errors='ignore')),
+                        files))
+    result_folder_name = os.path.join(OUTPUT_FOLDER, datetime.now().strftime("%Y_%m_%d-%H_%M_%S")+"_time_agg")
+    os.mkdir(result_folder_name)
+    results_df_dict = {}
+    for agg_name, agg_func in agg_funcs_dict.items():
+        columns = {}
+        for param_name, df in dfs_dict.items():
+            param_name = param_name.replace('.csv', '')
+            columns[param_name] = df.agg(agg_func, axis=1)
+        results_df_dict[agg_name] = pd.DataFrame.from_dict(columns)
+        results_df_dict[agg_name].to_csv(os.path.join(result_folder_name, f"{agg_name}.csv"))
+    return results_df_dict
+
+
+def plot_parameter_propagation_aggregated(mean_df: pd.DataFrame, std_df: pd.DataFrame,
```

Comment:
יש סיבה לתת את שני ה
DataFrames
ככה?
אולי לפחות שיהיה
API
שמניח שזה בעמודות של
mean וstd?
למשל:
```
def plot_parameter_aggregation(df: pd.DataFrame, parameter_names: List[str] = None):
   return plot_parameter_propagation_aggregated(df['mean'], df['std'], parameter_names)
```

Reply:
it plots for multiple parameters, 
to allow what you suggested the input needs to be 3 dimensional like numpy ndarray. but i prefer to keep if pandas dataframe.

Reply:
Wow. Good to know - I never realised that pandas doesn’t support n-dimensional datasets but only tabular datasets.